server/stream.go

```diff
@@ -4996,17 +5011,20 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 		}
 	}
 
+	// Find the message TTL if any.
+	ttl := getMessageTTL(hdr)
```

Comment:
Reject if ttl present and stream is not opted in.

Reply:
I don't think we can do this, otherwise we can run into a problem when we source/mirror messages with a TTL into another stream.

Reply:
We need special processing for streams and mirrors anyway that ignores (for ingest criteria) certain headers. This would be one added to the list.

Reply:
Will add that to the list of follow-up changes to make.